pynitrokey/fido2/client.py

```diff
@@ -337,16 +340,22 @@ def simple_secret(
 
         return output
 
-    def cred_mgmt(self, pin):
-        # anyways unused code @todo
-        # client = self.get_current_fido_client()
-        dev = nkfido2.find(serial)
-        client = dev.client
-        client_pin = ClientPin(dev.ctap2)
-        client_pin.change_pin(old_pin, new_pin)
-        token = client_pin.get_pin_token(pin)
-        ctap2 = Ctap2(self.get_current_hid_device())
-        return CredentialManagement(ctap2, client_pin.protocol, token)
+    def cred_mgmt(self, serial, pin):
+        device = nkfido2.find(serial)
+        client_pin = ClientPin(device.ctap2)
+
+        try:
+            client_token = client_pin.get_pin_token(pin)
+        except CtapError as e:
+            if str(CtapError.ERR.PIN_NOT_SET) in str(e):
+                local_critical("Please set a pin in order to manage credentials")
+            if str(CtapError.ERR.PIN_BLOCKED) in str(e):
+                local_critical(
+                    "Pin authentication has been blocked, try reinserting the key, setting a pin or "
+                    "reseting the device"
+                )
```

Comment:
According to the specification, `PIN_BLOCKED` should only be returned if the pin retry counter is zero, so a reset is required to fix it.  `PIN_AUTH_BLOCKED` is caused by three failing attempts in a row and can be fixed by reinserting the device.

Please update the message for `PIN_BLOCKED` and add a case for `PIN_AUTH_BLOCKED`.

Reply:
Changed the text and added the new case in the last commit.
I did notice that if no pin was set, re-plugging the device fixed both PIN_AUTH_BLOCKED and PIN_BLOCKED.
It seems like a niche case as mentioned before but I added it to the error message in case it does happen to someone.